2002-02-17  Andreas Schwab  <schwab@suse.de>

* signals.c (sigwait): Check for old sighandler being SIG_ERR,
	not NULL.

--- linuxthreads/signals.c.~1.23.~	Mon Jan 14 16:16:45 2002
+++ linuxthreads/signals.c	Sun Feb 17 00:51:41 2002
@@ -198,7 +198,7 @@
         s != __pthread_sig_cancel &&
         s != __pthread_sig_debug) {
       sigdelset(&mask, s);
-      if (sighandler[s].old == NULL ||
+      if (sighandler[s].old == (arch_sighandler_t) SIG_ERR ||
           sighandler[s].old == (arch_sighandler_t) SIG_DFL ||
           sighandler[s].old == (arch_sighandler_t) SIG_IGN) {
         sa.sa_handler = pthread_null_sighandler;

